R/drive_mime_type.R

```diff
@@ -0,0 +1,33 @@
+#' Lookup a mime type from a Google Drive type or file extension
+#'
+#' @param type Character. The Google Drive type or file extension you would
+#'   like to look up the mime type of.
+#' @template verbose
+#'
+#' @return Character. The mime type of input.
+#'
+#' @examples
+#' ## get the mime type for Google Spreadsheets
+#' drive_mime_type("spreadsheet")
+#'
+#' ## get the mime type for jpegs
+#' drive_mime_type("jpeg")
+#' @export
+drive_mime_type <- function(type = NULL, verbose = TRUE) {
+
+  if (!(is.character(type) && length(type) == 1)) {
+    stop("Please update `type` to be a character string.", call. = FALSE)
+  }
+
+  mime_type <- .drive$mime_tbl$mime_type[.drive$mime_tbl$mime_type == type |
+                                           (!is.na(.drive$mime_tbl$human_type) &
+                                              (.drive$mime_tbl$human_type == type))
+                                         ]
+  if (length(mime_type) == 0L) {
+    if (verbose) {
+      message(glue::glue("Ignoring `type` input. We do not have a mime type for files of type: {sq(type)}"))
+    }
+    return(invisible(NULL))
```

Comment:
I think the property that a real MIME type passes through unchanged is attractive. It's generally good if you could pass the output of `drive_mime_type()` through it again and not change anything.

But that doesn't mean you have to pass *anything* through. For input that doesn't match the human types, you could attempt a match against actual MIME types.

It's possible you should have another little helper that catches `NA`. Or you could let the API raise the error about a bad MIME type.